sycl/include/sycl/detail/pi.h

```diff
@@ -2352,6 +2445,222 @@ piextEnqueueCommandBuffer(pi_ext_command_buffer command_buffer, pi_queue queue,
+/// \param event_wait_list is the list of events to wait on before copying
+/// \param event is the returned event representing this operation
+__SYCL_EXPORT pi_result piextMemImageCopy(
+    pi_queue command_queue, void *dst_ptr, void *src_ptr,
+    const pi_image_format *image_format, const pi_image_desc *image_desc,
+    const pi_image_copy_flags flags, pi_image_offset src_offset,
+    pi_image_offset dst_offset, pi_image_region copy_extent,
+    pi_image_region host_extent, pi_uint32 num_events_in_wait_list,
+    const pi_event *event_wait_list, pi_event *event);
+
+/// API to query an image memory handle for specific properties.
+///
+/// \param mem_handle is the handle to the image memory
+/// \param param_name is the queried info name
+/// \param param_value is the returned query value
+/// \param param_value_size_ret is the returned query value size
+__SYCL_EXPORT pi_result piextMemImageGetInfo(
+    const pi_image_mem_handle mem_handle, pi_image_info param_name,
+    void *param_value, size_t *param_value_size_ret);
+
+/// API to import external memory in the form of a file descriptor.
+///
+/// \param context is the pi_context
+/// \param device is the pi_device
+/// \param size is the size of the external memory
+/// \param file_descriptor is the file descriptor
+/// \param ret_handle is the returned interop memory handle to the external
+/// memory
+__SYCL_EXPORT pi_result
+piextMemImportOpaqueFD(pi_context context, pi_device device, size_t size,
+                       int file_descriptor, pi_interop_mem_handle *ret_handle);
+
+/// API to map an interop memory handle to an image memory handle.
+///
+/// \param context is the pi_context
+/// \param device is the pi_device
+/// \param image_format format of the image (channel order and data type)
+/// \param image_desc image descriptor
+/// \param mem_handle is the interop memory handle to the external memory
+/// \param ret_mem is the returned image memory handle to the externally
+/// allocated memory
+__SYCL_EXPORT pi_result piextMemMapExternalArray(
+    pi_context context, pi_device device, pi_image_format *image_format,
+    pi_image_desc *image_desc, pi_interop_mem_handle mem_handle,
+    pi_image_mem_handle *ret_mem);
+
+/// API to destroy interop memory.
+///
+/// \param context is the pi_context
+/// \param device is the pi_device
+/// \param memory_handle is the handle to interop memory to be freed
+__SYCL_EXPORT pi_result piextMemReleaseInterop(
+    pi_context context, pi_device device, pi_interop_mem_handle memory_handle);
+
+/// API to import an external semaphore in the form of a file descriptor.
+///
+/// \param context is the pi_context
+/// \param device is the pi_device
+/// \param file_descriptor is the file descriptor
+/// \param ret_handle is the returned interop semaphore handle to the external
+/// semaphore
+__SYCL_EXPORT pi_result piextImportExternalSemaphoreOpaqueFD(
+    pi_context context, pi_device device, int file_descriptor,
```

Comment:
thanks @isaacault . My understanding is that we want this to be supported by L0 in the near future (few months). Since here we are defining the SYCL extension, shouldn't from the beginning account for these cases, instead of continuing changing it? @gmlueck , what do you suggest?

Reply:
It looks like the SYCL API is already defined in a way that works for both Linux and Windows, so the question here is about the PI layer.  Have you talked with the unified runtime (UR) team to see how they want to handle this?

Reply:
Are there other instances of using file descriptors in API's?

I'd be tempted to make this a platform dependent type in the UR header, e.g.

```cpp
#ifdef WIN32
typedef ur_file_descriptor_t HANDLE
#else
typedef ur_file_descriptor_t int
#endif
```

Reply:
thanks @gmlueck , @kbenzie .

I see, in the UR, we have it also defined as 32-bits (uint32_t bit actually). 

```
uint32_t fileDescriptor
```

so we have a mismatch here also, as here the PI is defining int and UR uint32_t. 

Suggestion from @kbenzie would work, and would be all that's needed if SYCL were calling UR directly, but since we still have the PI layer, would we need that defines also on this PR? Another approach is that for now, SYCL supports 32-bits (linux) handles, and once PI is removed completely, then Windows would be supported.

Reply:
We need to incorporate the changes from https://github.com/oneapi-src/unified-runtime/pull/716/files, right?